LayoutTests/inspector/timeline/timeline-event-performance-mark.html

```diff
@@ -0,0 +1,96 @@
+    InspectorTest.TimelineEvent.captureTimelineWithMarker = function({expression, markerType, timelineType}) {
+        let savePageDataPromise = InspectorTest.awaitEvent("SavePageData").then((event) => {
+            return event.data;
+        });
+
+        let promise = new WI.WrappedPromise;
+
+        let listener = WI.timelineManager.addEventListener(WI.TimelineManager.Event.CapturingStateChanged, (event) => {
+            if (WI.timelineManager.capturingState === WI.TimelineManager.CapturingState.Active) {
+                let recording = WI.timelineManager.activeRecording;
+                
+                let markerAddedListener = recording.addEventListener(WI.TimelineRecording.Event.MarkerAdded, (markerAddedEvent) => {
+                    let {marker} = markerAddedEvent.data;
+                    if (markerType && marker.type !== markerType)
+                        return;
+
+                    recording.removeEventListener(WI.TimelineRecording.Event.MarkerAdded, markerAddedListener);
+
+                    InspectorTest.log("Stopping Capture...");
+                    WI.timelineManager.stopCapturing();
+                });
+
+                InspectorTest.log("Evaluating...");
+                InspectorTest.evaluateInPage(expression)
+                .catch((error) => {
+                    promise.reject(error);
+                });
+                return;
+            }
+
+            if (WI.timelineManager.capturingState === WI.TimelineManager.CapturingState.Inactive) {
+                WI.timelineManager.removeEventListener(WI.TimelineManager.Event.CapturingStateChanged, listener);
+                InspectorTest.assert(savePageDataPromise, "savePageData should have been called in the page before capturing was stopped.");
+                savePageDataPromise.then((data) => {
+                    promise.resolve(data);
+                });
+                return;
+            }
+        });
+
+        InspectorTest.log("Starting Capture...");
+        const newRecording = true;
+        WI.timelineManager.startCapturing(newRecording);
+
+        return promise.promise;
+    }
+});
+
+function testPerformanceMark() {
+    let startTime = performance.now();
+    while (startTime + 10 > performance.now());
+    performance.mark('foo');
+    performance.mark('bar', {startTime});
+    savePageData({});
+}
+
+function test()
+{
+    let suite = InspectorTest.createAsyncSuite("TimelineEvent.PerformanceMark");
+
+    suite.addTestCase({
+        name: "TimelineEvent.PerformanceMark.performanceMark",
+        async test() {
+            let pageRecordingData = await InspectorTest.TimelineEvent.captureTimelineWithMarker({
+                expression: `testPerformanceMark()`,
+                markerType: WI.TimelineMarker.Type.TimeStamp,
+            });
+
+            let recording = WI.timelineManager.activeRecording;
+
+            InspectorTest.expectEqual(recording._exportDataMarkers.length, 2);
```

Comment:
We should probably rename `_exportDataMarkers` to `_markers` and add a public accessor for it (`get markers() { return _markers; }`) in `WI.TimelineRecording`. That would be a cleaner interface than reaching into its private properties.